### PR TITLE
Added scripts for TEALS mailing

### PR DESCRIPTION
**Pull request by aoby**
CC @ashercodeorg. Asher, do you want to be included in the test mails too?

@tanyaparker what is <%= cta %> in the template? Should I be populating that with something?

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Happy to be included, though I don't need to be.

File: bin/mail/2016-05-04-TEALS/generate-list.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+# Volunteer engineers in the US
+volunteer_query = %q(
+  kind_s: VolunteerEngineerSubmission2015 &&
+  experience_s: software_professional &&
+  volunteer_after_hoc_b: true &&
+  time_commitment_s: (weekly OR more) && (
+    create_ip_country_s: "United States" ||
+    location_country_code_s: "US"
+  )
```
Comment (ashercodeorg) [inline bin/mail/2016-05-04-TEALS/generate-list.rb:41]: This doesn't filter by age (e.g., 20 or older), per the spec. Should it?

File: bin/mail/2016-05-04-TEALS/generate-list.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  num_volunteers = state ? NUM_VOLUNTEERS_BY_STATE[state] : nil
+  email = result['email_s'].downcase.strip
+
+  results[email] = {
+    email: email,
+    name: result['name_s'],
+    state: state,
+    num_volunteers: num_volunteers
+  } unless UNSUBSCRIBERS[email]
+end
```
Comment (ashercodeorg) [inline bin/mail/2016-05-04-TEALS/generate-list.rb:64]: Is there a reason results is a hash of hashes rather than an array of hashes?

**Comment by aoby** [comment]
In the spec that's under Facebook Post. Email just has the solr query. @tanyaparker should we filter age too?

File: bin/mail/2016-05-04-TEALS/generate-list.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  'Indiana' => 14,
+  'Michigan' => 14,
+  'Minnesota' => 14,
+  'South Carolina' => 13,
+  'Louisiana' => 12,
+  'Colorado' => 10,
+  'Utah' => 9,
+  'Pennsylvania' => 8,
+  'District of Columbia' => 5
+}
```
Comment (ashercodeorg) [inline bin/mail/2016-05-04-TEALS/generate-list.rb:31]: Please add a comment indicating these numbers are arbitrary (e.g., based on need).

File: bin/mail/2016-05-04-TEALS/generate-list.rb (hunk 2 above)
Comment (aoby) [inline bin/mail/2016-05-04-TEALS/generate-list.rb:64]: To de-dupe by email. I copied and extended from [here](https://github.com/code-dot-org/code-dot-org/blob/93a719024a037590924cd051b862770f3b24dbb9/bin/mail/mailing-common/mailing-list-utils.rb#L75-L75).

File: bin/mail/2016-05-04-TEALS/generate-list.rb (hunk 3 above)
Comment (aoby) [inline bin/mail/2016-05-04-TEALS/generate-list.rb:31]: updated

File: bin/mail/2016-05-04-TEALS/generate-list.rb (hunk 2 above)
Comment (ashercodeorg) [inline bin/mail/2016-05-04-TEALS/generate-list.rb:64]: Nice. If that is the case, does it make sense to iterate in reverse order (via `reverse_each`) with the assumption that later entries will be more accurate for geo-location?

(not that it really matters)

Comment (aoby) [inline bin/mail/2016-05-04-TEALS/generate-list.rb:64]: Good call. Might as well. Updated.

File: bin/mail/2016-05-04-TEALS/generate-list.rb (hunk 1 above)
Comment (tanyaparker) [inline bin/mail/2016-05-04-TEALS/generate-list.rb:41]: The age filter was just for Facebook, not needed here.

**Comment by tanyaparker** [comment]
LGTM

**Comment by aoby** [comment]
I didn't realize name and email get stripped [here](https://github.com/code-dot-org/code-dot-org/blob/4939cf2ef354c58b6c4fd7081070ce270130f64e/bin/mail/send-to-mailing-list#L77-L77), so I'm adding another field name_s (copy of name) and changing that param in the email template so name actually shows up in the greeting.

**Pull request merged**